files/initramfs-tools/union-mount.j2

```diff
@@ -39,10 +39,38 @@ set_tmpfs_log_partition_size()
   [ $maxsize -le $varlogsize ] && varlogsize=$maxsize
 }
 
+whitelist_rw_folder()
+{
+  image_dir=$1
+  whitelist_file=${rootmnt}/host/$image_dir/whitelist_paths.conf
+  
+  # Return if the whitelist file does not exist
+  if ! test -f "${whitelist_file}"; then
+    return
+  fi
+
+  # Return if the secure_boot_enable option is not set
+  if ! (cat /proc/cmdline | grep -i -q "secure_boot_enable=[y1]"); then
+    return
+  fi
+
+  rw_dir=${rootmnt}/host/$image_dir/rw
+
+  # Set the grep pattern file, remove the blank line in config file
+  whitelist_pattern_file=${rootmnt}/host/$image_dir/whitelist_paths.pattern
+  awk -v rw_dir="$rw_dir" 'NF {print rw_dir"/"$0"$"}' ${whitelist_file} > $whitelist_pattern_file
+
+  # Find the files in the rw folder, and remove the files not in the whitelist
+  find ${rw_dir} -type f | grep -v -f $whitelist_pattern_file | xargs /bin/rm -f
+  rm -f $whitelist_pattern_file
+}
+
 ## Mount the overlay file system: rw layer over squashfs
 image_dir=$(cat /proc/cmdline | sed -e 's/.*loop=\(\S*\)\/.*/\1/')
 mkdir -p ${rootmnt}/host/$image_dir/rw
 mkdir -p ${rootmnt}/host/$image_dir/work
+## Whitelist rw folder
+whitelist_rw_folder "$image_dir"
```

Comment:
suggest name change to reflect the fact that files are removed.

Reply:
Changed to remove_not_whitelist_files